test/Microsoft.Net.Http.Headers.Tests/HeaderUtilitiesTest.cs

```diff
@@ -210,5 +210,29 @@ public void TryParseNonNegativeInt32_Succeeds(string valueString, long expected)
             Assert.True(HeaderUtilities.TryParseNonNegativeInt32(valueString, out value));
             Assert.Equal(expected, value);
         }
+
+        [Theory]
+        [InlineData("\"hello\\\\\"", "hello\\")]
+        [InlineData("\"hello\\\"\"", "hello\"")]
+        [InlineData("\"hello\\\\what\\\\\"", "hello\\what\\")]
+        [InlineData("hello\\\"foo\\\\bar\\\\baz\\\\", "hello\"foo\\bar\\baz\\")]
+        [InlineData("hello\\nfoo", "hello\\nfoo")]
```

Comment:
These two are unquoted strings containing non-token characters. Method should `throw` given such a value.